### PR TITLE
resolver: remove name resolver factory from ClientBuilder #176

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/resolver/URIResolvers.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      return Integer.MIN_VALUE;
+    }
+
+    @Override
+    public boolean supports(URI uri) {
+      return "http".equals(uri.getScheme()) || "https".equals(uri.getScheme());
+    }
+
+    @Override
+    public List<EquivalentAddressGroup> resolve(URI uri) {
```
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/resolver/URIResolvers.java]: should we also add a check on `uri` to see if it is supported in `resolve`? If not supported then don't add it the the group and cache. Right now it seems to me that the caller can by pass in whatever `uri` that it wants if it forgets to check the `uri` against `supports`.

**Comment by fanminshi** [comment]
looks good in general.

File: src/main/java/com/coreos/jetcd/resolver/URIResolvers.java (hunk 1 above)
Comment (lburgazzoli) [inline src/main/java/com/coreos/jetcd/resolver/URIResolvers.java]: The usage pattern would not require this double check but I'll add it for extra safety.

**Comment by lburgazzoli** [comment]
Done

File: src/main/java/com/coreos/jetcd/resolver/URIResolvers.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    @Override
+    public int priority() {
+      return Integer.MIN_VALUE;
+    }
+
+    @Override
+    public boolean supports(URI uri) {
+      return "http".equals(uri.getScheme()) || "https".equals(uri.getScheme());
+    }
```
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/resolver/URIResolvers.java]: we can use checking code in `isValidEndpointFormat` for Direct resolver's `supports`.

e.g

```java
private static final List<String> DRIECT_SCHEMES = Arrays.asList("http", "https");

public boolean supports(URI uri) {
    if (SCHEMES.stream().noneMatch(u.getProtocol()::equals)) {		
      return false;		
    }
    
    // endpoint must contain a port.		
    if (u.getPort() == -1) {		
      return false;		
    }
    
    // endpoint must not contain a path.		
    return u.getPath().isEmpty();	
}
```

File: src/test/java/com/coreos/jetcd/resolver/URIResolversTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    URIResolvers.Direct discovery = new URIResolvers.Direct();
+
+    for (String scheme : DIRECT_SCHEMES) {
+      URI uri = URI.create(scheme + "://127.0.0.1:2379");
+
+      assertThat(discovery.supports(uri)).isTrue();
+      assertThat(discovery.resolve(uri).size()).isGreaterThan(0);
+    }
+  }
+
```
Comment (fanminshi) [inline src/test/java/com/coreos/jetcd/resolver/URIResolversTest.java]: Can we add tests for invalid urls for Direct Resolvers.

transform those deleted tests to here.

```java
-  @Test(dataProvider = "invalidEndpoints", expectedExceptions = IllegalArgumentException.class)		
 -  public void testBuild_WithInValidFormatEndpoints(String endpoint) {		
 -    builder.setEndpoints(endpoint);		
 -  }		
 -		
 -  @DataProvider		
 -  private Object[][] invalidEndpoints() {		
 -    return new Object[][] {		
 -        {"://127.0.0.1:2379"},		
 -        {"mailto://127.0.0.1:2379"},		
 -        {"http://127.0.0.1"},		
 -        {"http://127.0.0.1:2379/path"}		
 -    };		
 -  
```

**Comment by fanminshi** [comment]
few concerns then lgtm.

**Comment by lburgazzoli** [comment]
Fixed now

**Pull request merged**